spec/module.dd

```diff
@@ -634,6 +633,27 @@ $(GNAME MixinDeclaration):
         $(GLINK DeclDefs), and is compiled as such.
         )
 
+$(H2 $(LEGACY_LNAME2 AllocatorRelic, allocator-relic, Allocator Relic))
```

Comment:
I don't see there any need to call it a relic, either in the grammar or documentation, unless `@disable new()` itself is up for deprecation.

Reply:
Sorry but I really like this rule name. from the Cambridge dictionnary

```d
relic
 noun
  /ˈrelik/
something left from a past time
```

That matches perfectly to what it is.

Reply:
By calling it a relic, to me this is implicitly saying "don't use", or at the very least discourages use (why would you use a relic feature in your codebase?)

Reply:
A relic is not something  you drop. It's cultural inheritage. It's something you keep because it tells something about history and culture.

Reply:
A relic conjures up that it is an artefact, vestige, or an archaism in the language.  But as far as I can tell, there is no reason to tell people that they should not use `@disable new()`.

Reply:
Suggest me another rule name. Discussing about semantics will lead to nothing, although, to be clear, I really liked my proposal.

Reply:
Maybe it's a cultural/language thing but I would read "relic" (in the context of software docs) as a soft deprecation.
(Read as "it exists but we don't dare to remove it").

Maybe use a neutral name based on the actual effect of `@disable new`? Smth. like `Disabling dynamic allocation`, ...

Reply:
Yes, I'm sorry, but I agree that relic is a terrible name for the reasons stated above.

Why not name it what it actually is. A few examples: "NewClassDisabler", "ClassAllocationOptOut", "DefaultClassAllocationRule".

Reply:
> Suggest me another rule name. Discussing about semantics will lead to nothing, although, to be clear, I really liked my proposal.

I think the following:
1. Use LEGACY_LNAME2 as you currently are.
2. Keep the grammar rule name as `Allocator`.
3. While the spec shouldn't be used to sell features, try to keep to strictly saying what the feature is, rather allude to what it was.

For instance (re-using some of the existing documentation):
> A class member function of the form
>
> ------
> @disable new();
> ------
>
> is used to explicitly disallow the $(D new) operator being used to allocate an object instance.
> For example, to enforce the usage of $(D std.allocators.make())  on classes or encourage
> stack allocation for structs.

I assume that `@disable new()` is permitted for structs too?